ci/Jenkinsfile-pyp

```diff
@@ -0,0 +1,205 @@
+pipeline {
+    agent any
+
+    parameters {
+        choice(
+            name: 'release_type',
+            choices: ['dev', 'stable'],
+            description: 'Indicate whether a dev or stable release should be built.'
+        )
+        choice(
+            name: 'version_increment',
+            choices: ['patch', 'minor', 'major'],
+            description: 'Indicates what kind of version bump should be done'
+        )
+        booleanParam(name: 'run_job', defaultValue: false, description: 'Allow for dry-runs')
+        booleanParam(name: 'upload', defaultValue: false, description: 'Publish to PyPi')
+    }
+
+    options {
+        disableConcurrentBuilds()
+        checkoutToSubdirectory('inmanta-sphinx')
+    }
+
+    stages {
+
+        stage("Setup venv and GitHub token") {
+            when {
+                expression { "${run_job}" == "true" }
+            }
+            steps {
+                dir('inmanta-sphinx') {
+                    sh '''
+                        python3 -m venv ${WORKSPACE}/env
+
+                        echo 'echo \${GITHUB_TOKEN}' > ${WORKSPACE}/ASK_PASS
+                        chmod u+x ${WORKSPACE}/ASK_PASS
+                    '''
+                }
+            }
+        }
+
+        stage("Setup next branch for major or minor release") {
+            when{
+                allOf {
+                    expression { "${run_job}" == "true" }
+                    expression { "${release_type}" == "stable" }
+                    anyOf {
+                        expression { "${version_increment}" == "major" }
+                        expression { "${version_increment}" == "minor" }
+                    }
+                }
+            }
+            steps{
+                dir('inmanta-sphinx') {
+                    withCredentials([string(credentialsId: 'fff7ef7e-cb20-4fb2-a93b-c5139463c6bf', variable: 'GITHUB_TOKEN')]) {
+                        sh '''
+                            export GIT_ASKPASS="${WORKSPACE}/ASK_PASS"
+                            ${WORKSPACE}/env/bin/python3 -m pip install -U pip bumpversion
+
+                            git checkout -b issue/version-bump
```

Comment:
We should make sure the correct branch is checked out. 
- For a minor or major release: master
- For a patch release: next